src/db/postgres/lrsql/postgres/sql/ddl.sql

```diff
@@ -281,6 +280,35 @@ CREATE TYPE scope_enum AS ENUM (
   'profile/read');
 ALTER TABLE IF EXISTS credential_to_scope ALTER COLUMN scope TYPE scope_enum USING (scope::scope_enum);
 
+/* Migration 2024-01-24 - Add profile and profile/read scopes */
+
+/* Simply adding the profile scope will cause a name clash with the reserved
+   OIDC profile scope. As a result, we add prefixes to create activity_profile
+   and agent_profile (and their read-only versions); this has the additional
+   benefit of narrowing the scope to just activity and agent profiles,
+   respectively. Since profile and profile/read have always been unused, we
+   are safe to remove them as enums. */
+
+-- :name alter-scope-enum-type-2!
+-- :command :execute
+-- :doc Add `activity_profile`, `activity_profile/read`, `agent_profile`, and `agent_profile/read` to `credential-to-scope.scope` enum.
+ALTER TABLE IF EXISTS credential_to_scope ALTER COLUMN scope TYPE VARCHAR(255);
+DROP TYPE IF EXISTS scope_enum;
+CREATE TYPE scope_enum AS ENUM (
+  'statements/write',
+  'statements/read',
+  'statements/read/mine', -- new
+  'all/read',
+  'all',
+  'state',
+  'state/read',
+  'define',
+  -- new; add document/profile scopes (also document/state out of being pedantic)
+  'activities_profile',
+  'activities_profile/read',
+  'agents_profile',
+  'agents_profile/read');
```

Comment:
Not sure which is better: the plural `activities_profile`/`agents_profile` or the singular `activity_profile`/`agent_profile`

Reply:
I would go with singular

Reply:
Actually I realized that these scope names follow from the names of their respective resource paths. So the `statements` scope matches the `/statements` resource, and the `state` scope matches the `/state` scope. By that logic, the `activities_profile` and `agents_profile` scopes should follow the `/activities/profile` and `/agents/profile` resources (and the hypothetical scopes `activities` and `agents` the resources `/activities` and `/agents`).

Reply:
yeah that justification is sound to me